stacks/api/function/enqueuer/src/database.ts

```diff
@@ -0,0 +1,95 @@
+import {ChangeStream, DatabaseService} from "@spica-server/database";
+import {DatabaseQueue, EventQueue} from "@spica-server/function/queue";
+import {Database, Event} from "@spica-server/function/queue/proto";
+import {Description, Enqueuer} from "./enqueuer";
+
+interface DatabaseOptions {
+  collection: string;
+  type: "INSERT" | "UPDATE" | "REPLACE" | "DELETE";
+}
+
+export class DatabaseEnqueuer extends Enqueuer<DatabaseOptions> {
+  description: Description = {
+    title: "Database",
+    name: "database",
+    icon: "view_agenda",
+    description: "Catch up the events happen in the database."
+  };
+
+  private streams = new Set<ChangeStream>();
+
+  constructor(
+    private queue: EventQueue,
+    private databaseQueue: DatabaseQueue,
+    private db: DatabaseService
+  ) {
+    super();
+  }
+
+  subscribe(target: Event.Target, options: DatabaseOptions): void {
+    const stream = this.db.collection(options.collection).watch(
+      [
+        {
+          $match: {operationType: options.type.toLowerCase()}
+        }
+      ],
+      {fullDocument: "updateLookup"}
+    );
+
+    stream.on("change", rawChange => {
+      const change = new Database.Change();
+      change.kind = getChangeKind(rawChange.operationType);
+      change.document = rawChange.fullDocument ? JSON.stringify(rawChange.fullDocument) : undefined;
```

Comment:
@thesayyn Does the "DELETE" option pass the deleted data or does it pass undefined to the function?

Reply:
It is null if the event type is DELETE.